### PR TITLE
Check at start of RDF run for size factor issues

### PR DESCRIPTION
**Pull request by rprospero**
This should close issue #954

### CODE REVIEW

File: src/modules/rdf/process.cpp [hunk 1]
```diff
@@ -48,6 +48,12 @@ bool RDFModule::process(Dissolve &dissolve, ProcessPool &procPool)
      * multiple independent Configurations, we must loop over the specified targetConfigurations_ and calculate the partials
      * for each.
      */
+
+    // Check that Configurations have unmodified size factor
+    if (std::all_of(targetConfigurations_.begin(), targetConfigurations_.end(),
+                    [](const auto *cfg) { return cfg->appliedSizeFactor() != 1.0; }))
```
Comment (trisyoungs) [inline src/modules/rdf/process.cpp:54]: We're comparing two `double`s here, which could be seen as risky - should we do a check on the difference between the applied factor and `1.0` instead?

File: src/modules/rdf/process.cpp [hunk 2]
```diff
@@ -48,6 +48,12 @@ bool RDFModule::process(Dissolve &dissolve, ProcessPool &procPool)
...
+        return true;
```
Comment (trisyoungs) [inline src/modules/rdf/process.cpp:55]: Printing a message here would be good and informative, I think:
```suggestion
    if (std::all_of(targetConfigurations_.begin(), targetConfigurations_.end(),
                    [](const auto *cfg) { return cfg->appliedSizeFactor() != 1.0; }))
    {
        Messenger::print("One or more configurations have an applied size factor, so RDF calculation will be skipped.\n");
        return true;
    }
```

**Review by trisyoungs — APPROVED** [review]
Two small comments (on three lines of code!).  This will achieve what we want in the immediate sense.  We should consider a more general implementation of this kind of functionality across all modules, for which I'll submit a new issue.

File: src/modules/rdf/process.cpp (hunk 1 above)
Comment (rprospero) [reply src/modules/rdf/process.cpp]: I've updated this to give us two free bits of slack.  That should handle most issues, but there's a 52 bit mantissa, so we might be able to afford a third bit.

**Pull request merged**